### ISSUE TITLE
Daemon user with /bin/false as shell does not work on CentOS 6.5

### ISSUE THREAD

**Issue by kodemaniak** [sbt/sbt-native-packager#515]
It seems that the daemon user is created with `/bin/false` as his default shell. For me this resulted in the service not being able to start. Starting the service by invoking the script under `/usr/share/<service>/bin` worked. 

I now override the `%pre` scriptlet and create the user with `/bin/bash` as his shell. That seems to work. I am no expert with writing init.d scripts, nohup or runuser, but it seems that `/bin/false` is not working correctly on every system. I have other daemon users on my system that use `/bin/nologin`.

FYI, I tested it on a vagrant box using the `chef/centos-6.5` image.

**Comment by muuki88** [comment]
That's true, the users are created with `/bin/false` as a shell is only required if you want to login as a user.  However I'm not too familiar with CentOS (didn't know that there's a `/bin/nologin`).

Can you post the exact error what is happening?

There is also a way to configure the shell used, so you don't have to extend the `%pre` scriptlet.
You should set this with

```scala
daemonShell in Linux := "/bin/nologin"
```

There are a [bunch of other linux related settings](https://github.com/sbt/sbt-native-packager/blob/master/src/main/scala/com/typesafe/sbt/packager/linux/LinuxPlugin.scala#L60-L67).

- [Daemon User Shell Setting](https://github.com/sbt/sbt-native-packager/blob/master/src/main/scala/com/typesafe/sbt/packager/linux/Keys.scala#L15)
- [Add User Method](https://github.com/sbt/sbt-native-packager/blob/master/src/main/resources/com/typesafe/sbt/packager/archetypes/java_server/rpm/preinst-template#L6)
- [Add User Method Declaration](https://github.com/sbt/sbt-native-packager/blob/master/src/main/resources/com/typesafe/sbt/packager/linux/control-functions#L10)

**Comment by kardapoltsev** [comment]
I check other daemons on some machines:
Gentoo: mostly /sbin/nologin
Debian stable: mostly /bin/false, few /usr/sbin/nologin
Ubuntu 14.04: mostly /usr/sbin/nologin

But all of them have /bin/false. I think it's reasonable default for us.

**Comment by kodemaniak** [comment]
Thanks for the tip with `daemonShell`, I tried to figure out if there is some setting for this, but did not look into `Keys.scala`.

To me it seems that not `/bin/false` itself is the problem, but rather the combination with `runuser`. Some googling revealed that at least `su` requires a real shell to work, and `runuser` is probably very similar (and does even fall back to `su` semantics in certain cases regarding the man page).

Whe I look at the init scripts of some of the services that indeed have `/bin/false` as the shell for their user, the seem to use `start-stop-daemon` (Ubuntu) to daemonize the process.

**Comment by muuki88** [comment]
That could be a reason. Unfortunately the `start-stop-daemon` facilities on non Debian systems are... Let's say tricky.

If you have a good solution for rpm initv scripts, we are happy to merge :)

**Comment by kodemaniak** [comment]
I always try to avoid writing init scripts myself ;) I have other Java daemons on my system that use `su --shell`. When I change the `sbt-native-packager` generated init script to use

````
nohup su $DAEMON_USER -s "/bin/sh" -c "${RUN_CMD}" >> /dev/null 2>&1 &
````

the service starts even with `/bin/false` as the shell.

**Comment by hobbsm** [comment]
We've got the same issue and based on this report we've added the following rpmPost command to work around it:

sed -i 's@nohup runuser $DAEMON_USER@nohup su $DAEMON_USER -s "/bin/sh" -c@g' /etc/init.d/yourprogram

Would be great to get rid of this soon though!

**Comment by muuki88** [comment]
Actually we have. @dwhjames did an amazing job fixing the rpm scripts. see 45bc5dbaffdca8eaef4b498c6b93e3e03350428d 

This is already included in the `1.0.0` release
